File: sdks/python/tests/test_fingerprint.py
"""Tests for fingerprint.py — fingerprint function."""

from autonoma.fingerprint import fingerprint


class TestFingerprint:
    def test_produces_16_char_hex_string(self):
        result = fingerprint({"key": "value"})
        assert len(result) == 16
        int(result, 16)  # valid hex

    def test_order_independent_for_object_keys(self):
        fp1 = fingerprint({"a": 1, "b": 2})
        fp2 = fingerprint({"b": 2, "a": 1})
        assert fp1 == fp2

    def test_different_data_produces_different_fingerprints(self):
        fp1 = fingerprint({"key": "value1"})
        fp2 = fingerprint({"key": "value2"})
        assert fp1 != fp2

    def test_handles_nested_objects(self):
        result = fingerprint({"outer": {"inner": "deep"}})
        assert len(result) == 16

    def test_handles_arrays(self):
        result = fingerprint([1, 2, 3])
        assert len(result) == 16

    def test_handles_strings(self):
        result = fingerprint("hello")
        assert len(result) == 16

    def test_handles_numbers(self):
        result = fingerprint(42)
        assert len(result) == 16

File: sdks/ruby/test/test_fingerprint.rb
# frozen_string_literal: true

require "minitest/autorun"
require_relative "../lib/autonoma"

class TestFingerprint < Minitest::Test
  def test_produces_16_char_hex
    fp = Autonoma::Fingerprint.fingerprint({ "name" => "test" })
    assert_equal "7d9fd2051fc32b32", fp
    assert_match(/\A[a-f0-9]{16}\z/, fp)
  end

  def test_order_independent_for_object_keys
    fp1 = Autonoma::Fingerprint.fingerprint({ "z" => 1, "a" => 2, "m" => 3 })
    fp2 = Autonoma::Fingerprint.fingerprint({ "a" => 2, "m" => 3, "z" => 1 })
    assert_equal fp1, fp2
    assert_equal "ebba85cfdc0a724b", fp1
  end

  def test_different_values_different_fingerprints
    fp1 = Autonoma::Fingerprint.fingerprint({ "a" => 1 })
    fp2 = Autonoma::Fingerprint.fingerprint({ "a" => 2 })
    refute_equal fp1, fp2
  end

  def test_handles_arrays
    fp = Autonoma::Fingerprint.fingerprint([1, 2, 3])
    assert_equal "a615eeaee21de517", fp
  end

  def test_handles_strings
    fp = Autonoma::Fingerprint.fingerprint("hello")
    assert_equal "5aa762ae383fbb72", fp
  end

  def test_handles_numbers
    fp = Autonoma::Fingerprint.fingerprint(42)
    assert_equal "73475cb40a568e8d", fp
  end

  def test_simple_object
    fp = Autonoma::Fingerprint.fingerprint({ "a" => 1 })
    assert_equal "015abd7f5cc57a2d", fp
  end
end
